sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c

```diff
@@ -347,7 +347,11 @@ uint8_t gnrc_sixlowpan_nd_opt_ar_handle(kernel_pid_t iface, ipv6_hdr_t *ipv6,
                 /* TODO: multihop DAD behavior */
                 uint16_t reg_ltime;
                 if (nc_entry == NULL) {
-                    if ((nc_entry = gnrc_ipv6_nc_add(iface, &ipv6->src, sl2a, sl2a_len,
+                    /* The address provided by ARO is always EUI-64,
+                     * regardless of link-layer address.
+                     * https://tools.ietf.org/html/rfc6775#section-4.1 */
+                    if ((nc_entry = gnrc_ipv6_nc_add(iface, &ipv6->src,
+                                                     &ar_opt->eui64, sizeof(eui64_t),
```

Comment:
Errm this entry will be used for address translation, too (so with this patch short addresses won't be used as destination addresses). The EUI-64 to *identify* the node in address registration is set in https://github.com/RIOT-OS/RIOT/pull/4446/files#diff-9b0aedc5fd4543061c862e245fe43c75R359 and checked in https://github.com/RIOT-OS/RIOT/pull/4446/files#diff-9b0aedc5fd4543061c862e245fe43c75R337